### PR TITLE
statsd listens on TCP for Windows

### PR DESCRIPTION
**Pull request by palourde**
Signed-off-by: Simon Plourde <simon@sensu.io>

## What is this change?

The `gostatsd` pkg tries to use the `ReadBatch` method, which is not implemented on Windows, and ultimately provokes an out-of-memory error. Instead, we force a TCP connection on Windows in order to use gostatsd's `GenericBatchReader`, which uses `ReadFrom` instead 
(https://github.com/atlassian/gostatsd/blob/8eecbec4c92a443f58888bdddd31d3fc66114117/pkg/statsd/batched_reader.go#L64).

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/1402

## Does your change need a Changelog entry?

Added!

## Do you need clarification on anything?

Non!

## Were there any complications while making this change?

Non!

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -382,7 +384,19 @@ func (a *Agent) Run() error {
 
 	if !a.config.StatsdServer.Disable {
 		logger.Info("starting statsd server on address: ", a.statsdServer.MetricsAddr)
-		go a.statsdServer.Run(a.context)
+
+		// We need to force a TCP connection for Windows. See
+		// https://github.com/sensu/sensu-go/issues/1402
+		if runtime.GOOS == "windows" {
+			conn, err := net.ListenPacket("tcp", a.statsdServer.MetricsAddr)
+			socketFactory := func() (net.PacketConn, error) {
```
[the remark this answers is not in the document]
Comment (echlebek) [reply agent/agent.go]: Oh, also I think you can force IPv4 here,
```
conn, err := net.ListenPacket("tcp4", ...)
```

[the remark this answers is not in the document]
Comment (palourde) [reply agent/agent.go]: Ah right, nvm, we should also use that logic: https://github.com/atlassian/gostatsd/blob/99f7a39855fc860055264fcec1cbae28bdc2deeb/pkg/statsd/statsd.go#L64-L73

[the remark this answers is not in the document]
Comment (echlebek) [reply agent/agent.go]: Actually, I think you were right about this the first time.

**Comment by palourde** [comment]
Tests fails because of https://github.com/sensu/sensu-go/pull/1439. Was working before merging master so I'll just merge!

**Pull request merged**